recipes/stdgpu/all/conanfile.py

```diff
@@ -0,0 +1,207 @@
+        "enable_contract_checks": None,
+        "use_32_bit_index": True,
+        "openmp": "llvm",
+    }
+
+    @property
+    def _min_cppstd(self):
+        if self.version == "1.3.0":
+            return 14
+        else:
+            return 17
+
+    @property
+    def _compilers_min_versions(self):
+        if self.version == "1.3.0":
+            # Based on https://github.com/stotko/stdgpu/tree/1.3.0#building
+            return {
+                "gcc": "7",
+                "clang": "6",
+                "apple-clang": "6",
+                "msvc": "192",
+                "Visual Studio": "16",
+            }
+        else:
+            # > 1.3.0
+            # Based on https://github.com/stotko/stdgpu/tree/32e0517#building
+            return {
+                "gcc": "9",
+                "clang": "10",
+                "apple-clang": "12",
+                "msvc": "192",
+                "Visual Studio": "16",
+            }
+
+    def export_sources(self):
+        copy(self, "cmake/*", dst=self.export_sources_folder, src=self.recipe_folder)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.options.backend != "openmp":
+            del self.options.openmp
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        if self.options.backend == "cuda":
+            self.options["thrust"].device_system = "cuda"
+        elif self.options.backend == "openmp":
+            self.options["thrust"].device_system = "omp"
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        if self.options.backend != "hip":
+            self.requires("thrust/1.17.2", transitive_headers=True)
+        else:
+            # The baseline Thrust version provided by Nvidia and Conan is not compatible with HIP.
+            self.output.warning("HIP support requires Thrust with ROCm. "
+                                "Using Thrust from system instead of Conan.")
+        if self.options.backend == "openmp":
+            if self.options.openmp == "llvm":
+                self.requires("llvm-openmp/12.0.1", transitive_headers=True, transitive_libs=True)
+            else:
+                self.output.info("Using OpenMP backend with system OpenMP")
+
+    def build_requirements(self):
+        if Version(self.version) > "1.3.0":
+            self.tool_requires("cmake/[>=3.18 <4]")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_min_versions.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
```

Comment:
```suggestion
        if self.options.backend == "cuda" and self.dependencies["thrust"].options.device_system != "cuda":
            raise ConanInvalidConfiguration(f"{self.ref} option backend=cuda should use '-o thrust/*:device_system=cuda' as well.")
        if self.options.backend == "openmp" and self.dependencies["thrust"].options.device_system != "omp":
            raise ConanInvalidConfiguration(f"{self.ref} option backend=openmp should use '-o thrust/*:device_system=omp' as well.")

```

Forcing an option does not impose user's options, so you still need to validate it to avoid override.

Reply:
Good to know, thanks!